tests/unit/http.test.ts

```diff
@@ -1,28 +1,80 @@
+import * as url from "url";
+
+describe("url module import syntax works", () => {
+  it("global URL and imported URL are equal", () => {
+    const testUrl = "https://www.example.com";
+    const moduleUrl = new url.URL(testUrl);
+    const globalUrl = new URL(testUrl);
+    expect(moduleUrl).toEqual(globalUrl);
+  });
+  it("global URLSearchParams and imported URLSearchParams are equal", () => {
+    const paramsString = "topic=api&a=1&a=2&a=3";
+    const moduleSearchParams = new url.URLSearchParams(paramsString);
+    const globalSearchParams = new URLSearchParams(paramsString);
+    expect(moduleSearchParams).toEqual(globalSearchParams);
+  });
+  describe("import { URL } from 'url';", () => {
+    it("should parse a url hostname", () => {
+      const testUrl = new url.URL("https://www.example.com");
+      expect(testUrl.protocol).toEqual("https:");
+      expect(testUrl.host).toEqual("www.example.com");
+      expect(testUrl.hostname).toEqual("www.example.com");
+    });
+    it("toString method works", () => {
+      const testUrl = new url.URL("/base", "https://www.example.com");
+      expect(testUrl.toString()).toEqual("https://www.example.com/base");
+    });
+    it("canParse method works", () => {
+      const validCanParse = url.URL.canParse("https://www.example.com");
+      const invalidCanParse = url.URL.canParse("not_valid");
+      expect(validCanParse).toEqual(true);
+      expect(invalidCanParse).toEqual(false);
+      expect(url.URL.canParse("/foo", "https://example.org/")).toEqual(true);
+    });
+  });
+
+  describe("import { URLSearchParams } from 'url';", () => {
+    it("supports URLSearchParams basic API", () => {
+      const paramsString = "topic=api&a=1&a=2&a=3";
+      const searchParams = new url.URLSearchParams(paramsString);
+      searchParams.append("foo", "bar");
+      expect(searchParams.has("topic")).toBeTruthy();
+      expect(searchParams.has("foo")).toBeTruthy();
+      searchParams.delete("foo");
+      expect(searchParams.has("foo")).toBeFalsy();
+      expect(searchParams.get("topic")).toEqual("api");
+      expect(searchParams.getAll("a")).toEqual(["1", "2", "3"]);
+      searchParams.set("topic", "node");
+      expect(searchParams.get("topic")).toEqual("node");
+    });
+  });
+});
+
```

Comment:
Main change in this CR, the rest are linting formatting changes generated by prettier.